### DISCUSSION TITLE
Change behaviour of iOS DatePicker

### DISCUSSION THREAD

**Discussion by assurasc** [unoplatform/uno#13670] [General]
I'm trying to change the behaviour of the iOS date picker. I want it to use the new (since iOS14) style via the .inline datePickerStyle. It seems to use the older .wheel style at the moment.

On Xamarin.Forms I would be able to use a custom renderer for this sort of thing. I can't find an alternative in Uno. 

**Update**

Below is the code in Master branch - it would appear to set the style to wheels for iOS15 and newer?

```
DatePickerSelector.iOS.cs

private void UpdatePickerStyle()
{
	if (_picker == null)
	{
		return;
	}

	if (UIDevice.CurrentDevice.CheckSystemVersion(15, 0))
	{
		_picker.PreferredDatePickerStyle = UIDatePickerStyle.Wheels;

		return;
	}

	if (UIDevice.CurrentDevice.CheckSystemVersion(13, 4))
	{
		_picker.PreferredDatePickerStyle = FeatureConfiguration.DatePicker.UseLegacyStyle
									? UIDatePickerStyle.Wheels
									: UIDatePickerStyle.Inline;
	}
}
```

**Comment by assurasc** [comment]
I see that Apple revived the wheel picker in iOS 15 (hence the code update), but with a feature where you can tap the wheel to edit the values. The wheel picker we get does not have that.

How about we allow a legacy style on iOS 15 and above like for iOS 13.4 and above? i.e.

```
if (UIDevice.CurrentDevice.CheckSystemVersion(15, 0))
{
  _picker.PreferredDatePickerStyle = FeatureConfiguration.DatePicker.UseLegacyStyle
							  ? UIDatePickerStyle.Inline
							  : UIDatePickerStyle.Wheels;
  return;
}
```

**Reply by jeromelaban** [reply]
@ajpinedam may have additional insights there.
